scripts/zones/Den_of_Rancor/mobs/Ogama.lua

```diff
@@ -5,6 +5,25 @@
 ---@type TMobEntity
 local entity = {}
 
+entity.onMobFight = function(mob, target)
+    local mobHPP = mob:getHPP()
+    local appliedDmgBoost = mob:getLocalVar('appliedDmgBoost')
+
+    if
+        mobHPP <= 50 and
+        appliedDmgBoost == 0
+    then
+        mob:addMod(xi.mod.MAIN_DMG_RATING, mob:getMainLvl() + 2)
+        mob:setLocalVar('appliedDmgBoost', 1)
+    elseif
+        mobHPP > 50 and
+        appliedDmgBoost == 1
+    then
+        mob:delMod(xi.mod.MAIN_DMG_RATING, mob:getMainLvl() + 2)
+        mob:setLocalVar('appliedDmgBoost', 0)
+    end
+end
```

Comment:
Cool two-way latch 👍

Reply:
Why no simply set (not add, set) new dmg rating when under 50% hp, set original dmg rating in onMobSpawn and be done with the local var, simplifying onMobFight logic?

onMobFight is run per tick, so we want it as simple as possible

Reply:
> Why not simply set (not add, set) new dmg rating when under 50% hp, set original dmg rating in onMobSpawn and be done with the local var, simplifying onMobFight logic?
> 
> onMobFight is run per tick, so we want it as simple as possible

If the mob heals back above 50% while fighting or roaming then I think we want the boost to be removed, thus the need for the check.

Reply:
The mob has no regen nor any healing mechanics whatsoever though

Reply:
> The mob has no regen nor any healing mechanics whatsoever though. As easy as placing it onMobRoam aswell

Yeah but if we later find out that some mobskill it has does add regen on retail then the logic becomes incorrect. You have a good point about simplicity but I guess just my preference to make it more robust at the cost of more checks.